src/rdkafka_broker.c

```diff
@@ -3407,6 +3407,8 @@ rd_kafka_broker_op_serve(rd_kafka_broker_t *rkb, rd_kafka_op_t *rko) {
                                 : (topic_err
                                        ? topic_err
                                        : RD_KAFKA_RESP_ERR__UNKNOWN_PARTITION));
+
+                        rd_kafka_toppar_purge_internal_fetch_queue_maybe(rktp);
```

Comment:
Shall we do this only for the Consumer?

Reply:
Yes, better to avoid it for the producer given `rktp_fetchq` is allocated there too but the queue isn't forwarded we avoid looping through this empty queue.